### PR TITLE
Delaware standard deduction

### LINKED ISSUE #2377: Delaware standard deduction
reference:https://revenuefiles.delaware.gov/2022/PIT-RES_TY22_2022-02_Instructions.pdf

### PR DESCRIPTION
**Pull request by shuangding415**
### Summary
📄🧪📊

This pull request adds the functionality for the Delaware standard deduction in the policy engine. It includes the parameter, variable, and test files for the `de_standard_deduction` variable.

> _`de_standard_deduction`_
> _New parameter and variable_
> _Autumn leaves fall_

### Walkthrough
*  Add the Delaware standard deduction parameter, variable, and tests ([link](https://github.com/PolicyEngine/policyengine-us/pull/2379/files?diff=unified&w=0#diff-9088e519421ac20d9f4ea50a9f1055635cc9d8e20d55c5e390cc05bc58a4ec19R1-R19), [link](https://github.com/PolicyEngine/policyengine-us/pull/2379/files?diff=unified&w=0#diff-1202d2765326825e09c8fbb22faaf06364599649d7e5bba9b9bef605306e047eL1-R14), [link](https://github.com/PolicyEngine/policyengine-us/pull/2379/files?diff=unified&w=0#diff-737cadac1570131bde936015fd233be2e32b7c1881fb0fcf57484aa1fa0d0a34R1-R17))
  - Create the parameter file `standard.yaml` with the deduction amounts and metadata ([link](https://github.com/PolicyEngine/policyengine-us/pull/2379/files?diff=unified&w=0#diff-9088e519421ac20d9f4ea50a9f1055635cc9d8e20d55c5e390cc05bc58a4ec19R1-R19))
  - Define the variable `de_standard_deduction` in `standard.py` with the logic and metadata ([link](https://github.com/PolicyEngine/policyengine-us/pull/2379/files?diff=unified&w=0#diff-737cadac1570131bde936015fd233be2e32b7c1881fb0fcf57484aa1fa0d0a34R1-R17))
  - Add two test cases in `standard.yaml` to verify the output of the variable ([link](https://github.com/PolicyEngine/policyengine-us/pull/2379/files?diff=unified&w=0#diff-1202d2765326825e09c8fbb22faaf06364599649d7e5bba9b9bef605306e047eL1-R14))

fixes #2377

### CODE REVIEW

File: policyengine_us/parameters/gov/states/de/tax/income/deductions/standard.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    - filing_status
+JOINT:
+  2022-01-01: 6_500
+SINGLE:
+  2022-01-01: 3_250
+SEPARATE:
+  2022-01-01: 3_250
+HEAD_OF_HOUSEHOLD:
+  2022-01-01: 3_250
+
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/de/tax/income/deductions/standard.yaml:19]: need to add the widow case

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: run the `git pull upstream master` to fix last test

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
legal code ref please

File: policyengine_us/parameters/gov/states/de/tax/income/deductions/standard.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/de/tax/income/deductions/standard.yaml:1]: can you add the historic values (before 1999 and 1999-200) to the yaml file

File: policyengine_us/parameters/gov/states/de/tax/income/deductions/standard.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      href: https://revenuefiles.delaware.gov/2022/PIT-RES_TY22_2022-02_Instructions.pdf
+    - title: The Delaware Code - Title 30 - Chapter 11 - Subchapter II - § 1102
+      href: https://delcode.delaware.gov/title30/c011/sc02/index.html
+  label: Delaware Standard Deduction
+  unit: currency-USD
+  period: year
+  breakdown:
+    - filing_status
+JOINT:
+  2022-01-01: 6_500
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/de/tax/income/deductions/standard.yaml:14]:
```suggestion
  2000-01-01: 6_500
```

File: policyengine_us/parameters/gov/states/de/tax/income/deductions/standard.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+SINGLE:
+  2022-01-01: 3_250
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/de/tax/income/deductions/standard.yaml:16]:
```suggestion
  2000-01-01: 3_250
```

File: policyengine_us/parameters/gov/states/de/tax/income/deductions/standard.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+SEPARATE:
+  2022-01-01: 3_250
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/de/tax/income/deductions/standard.yaml:18]:
```suggestion
  2000-01-01: 3_250
```

File: policyengine_us/parameters/gov/states/de/tax/income/deductions/standard.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+HEAD_OF_HOUSEHOLD:
+  2022-01-01: 3_250
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/de/tax/income/deductions/standard.yaml:20]:
```suggestion
  2000-01-01: 3_250
```

File: policyengine_us/parameters/gov/states/de/tax/income/deductions/standard.yaml [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+WIDOW:
+  2022-01-01: 3_250
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/de/tax/income/deductions/standard.yaml:22]:
```suggestion
  2000-01-01: 3_250
```

File: policyengine_us/parameters/gov/states/de/tax/income/deductions/standard.yaml
Comment (shuangding415) [reply policyengine_us/parameters/gov/states/de/tax/income/deductions/standard.yaml:1]: done

File: policyengine_us/parameters/gov/states/de/tax/income/deductions/standard.yaml (hunk 2 above)
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/de/tax/income/deductions/standard.yaml:14]:
```suggestion
  2000-01-01: 6_500
```

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/de/tax/income/deductions/standard.yaml:14]: and all others

File: policyengine_us/parameters/gov/states/de/tax/income/deductions/standard.yaml [hunk 7]
```diff
@@ -0,0 +1,22 @@
+description: Delaware provides these standard deduction amounts.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/de/tax/income/deductions/standard.yaml:1]:
```suggestion
description: Delaware provides this standard deduction based on filing status.
```

File: policyengine_us/parameters/gov/states/de/tax/income/deductions/standard.yaml [hunk 8]
```diff
@@ -0,0 +1,22 @@
+description: Delaware provides this standard deduction based on filing status.
+metadata:
+  reference:
+    - title: Delaware individual income tax instructions for 2022
+      href: https://revenuefiles.delaware.gov/2022/PIT-RES_TY22_2022-02_Instructions.pdf
+    - title: The Delaware Code - Title 30 - Chapter 11 - Subchapter II - § 1102
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/de/tax/income/deductions/standard.yaml:6]:
```suggestion
    - title: The Delaware Code - Title 30 - Chapter 11 - Subchapter II - § 1109 - (a) - (3)
```

File: policyengine_us/parameters/gov/states/de/tax/income/deductions/standard.yaml [hunk 9]
```diff
@@ -0,0 +1,22 @@
+description: Delaware provides this standard deduction based on filing status.
+metadata:
+  reference:
+    - title: Delaware individual income tax instructions for 2022
+      href: https://revenuefiles.delaware.gov/2022/PIT-RES_TY22_2022-02_Instructions.pdf
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/de/tax/income/deductions/standard.yaml:5]: add page # to link

File: policyengine_us/variables/gov/states/de/tax/income/deductions/de_standard_deduction.py [hunk 10]
```diff
@@ -0,0 +1,16 @@
+from policyengine_us.model_api import *
+
+
+class de_standard_deduction(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Delaware standard deduction"
+    unit = USD
+    definition_period = YEAR
+    reference = "https://revenuefiles.delaware.gov/2022/PIT-RES_TY22_2022-02_Instructions.pdf"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/de/tax/income/deductions/de_standard_deduction.py:10]:
```suggestion
    reference = "https://delcode.delaware.gov/title30/c011/sc02/index.html"
```

File: policyengine_us/tests/policy/baseline/gov/states/de/tax/income/deductions/de_standard_deduction.yaml
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/de/tax/income/deductions/de_standard_deduction.yaml:1]: add a test for each filing status

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**